03-task-manager/starter/controllers/tasks.js

```diff
@@ -0,0 +1,41 @@
+const Task = require("../models/Task");
+
+const getAllTasks = async (req, res) => {
+  const tasks = await Task.find({});
+  res.status(200).json({ tasks });
+};
+
+const createTask = async (req, res) => {
+  const task = await Task.create(req.body);
+  res.status(201).json({ task });
+};
+
+const getTask = async (req, res) => {
+  const { id: taskID } = req.params;
+  const task = await Task.findOne({ _id: taskID });
+  res.status(200).json({ task });
+};
+
+const updateTask = async (req, res) => {
+  const { id: taskID } = req.params;
+
+  const task = await Task.findOneAndUpdate({ _id: taskID }, req.body, {
```

Comment:
One suggestion here: you could check if the task exists, check if the task is null, and if it does not exist, return a 404 error indicating that the task was not found.